### ISSUE TITLE
XR500 Invalid response code from router: 501

### ISSUE THREAD

**Issue by sebastien-ggp** [gruijter/com.gruijter.netgear#51]
Hello,
First of all thank you very much for your work. I have configured the X500 in router mode but I cannot get any information back? Thank you in advance

Error: Invalid response code from router: 501

{Firmware:V2.3.2.66WW,RegionTag:XR500_WW,Region:ww,Model:XR500,InternetConnectionStatus:Up,ParentalControlSupported:0,SOAPVersion:3.18,ReadyShareSupportedLevel:13,XCloudSupported:1,LoginMethod:2.0,host:10.0.0.1,port:80}

========== STARTING TEST ==========
Node version: v12.16.1
Netgear package version: 4.2.0
OS: linux 4.14.15-g3c513a6eca
Info: Homey fw:4.2.0 app: 3.2.3
t = 0
trying to auto discover Netgear routers...
{SpeedTest:2.0,GuestNetworkSchedule:1.0,TCAcceptance:1.0,SmartConnect:2.0,AttachedDevice:2.0,NameNTGRDevice:1.0,MaxMonthlyTrafficLimitation:4095000000}
t = 1.029
trying to get the LAN configuration...
{NewLANSubnet:255.255.255.0,NewWANLAN_Subnet_Match:1,NewLANMACAddress:CC40D0XXXX,NewLANIP:10.0.0.1,NewDHCPEnabled:true}
t = 1.143
trying to get the WAN configuration...
{NewEnable:1,NewConnectionType:Static,NewExternalIPAddress:192.168.1.252,NewSubnetMask:255.255.255.0,NewAddressingType:Static,NewDefaultGateway:192.168.1.254,NewMACAddress:CC40DXXXXXNewMACAddressOverride:0,NewMaxMTUSize:1500,NewDNSEnabled:1,NewDNSServers:192.168.1.254 1.1.1.1 8.8.8.8}
t = 1.433
trying to get the Traffic Meter Enabled Status...
Traffic Meter Enabled: true
t = 1.883
trying to get trafficMeter...
{newTodayUpload:13.68,newTodayDownload:153.71,newMonthUpload:505.83,newMonthDownload:8797}
t = 2.174
trying to get the Traffic Meter Options...
{newControlOption:No Limit,newNewMonthlyLimit:0,restartHour:0,restartMinute:0,restartDay:1}
t = 2.581
trying to get System Uptime (hh:mm:ss)...
1:05:14
t = 2.781
trying to get the Device configuration...
{BlankState:0,NewBlockSiteEnable:2,NewBlockSiteName:2,NewTimeZone:+1GMT,M3.5.0/2:00,M10.5.0/2:00,NewDaylightSaving:.5.0/2:00,M10.5.0/2:00}
t = 3.133
trying to get Parental Control Status...
404 Not Found. The requested function/page is not available
Parental Control Enabled: [object Object]
t = 3.898
trying to get Qos Status...
404 Not Found. The requested function/page is not available
Qos Enabled: [object Object]
t = 4.273
trying to get Qos Bandwidth options...
404 Not Found. The requested function/page is not available
{}
t = 4.575
trying to get available and selected WiFi channels...
Available channels 2.4G-1: Auto,1,2,3,4,5,6,7,8,9,10,11,12,13
selected channel 2.4G-1: Auto
Available channels 5.0G-1: 36,40,44,48,52,56,60,64,100,104,108,112,116,120,124,128
selected channel 5.0G-1: 44
5.0G-2 channels are not available
5.0G-2 channel is not available
t = 16.806
trying to get Smart Connect Status...
Smart Connect Enabled: true
t = 17.611
trying to get router logs...
last log: [site allowed: 192.168.1.137] from source 10.0.0.25, Sunday, December 20, 2020 01:03:19
t = 17.91
trying to get Guest Wifi Status...
2.4G-1 Guest wifi enabled: true
5.0G-1 Guest wifi enabled: false, method: 1
5.0G-2 Guest wifi is not available
t = 19.254
trying to get Device Access Control Status...
Block Device Enabled: true
t = 19.539
trying to get attachedDevices method 1...
Invalid response code from router: 501
{lastResponse:<?xml version=\1.0\ encoding=\UTF-8\?>\r\n<SOAP-ENV:Envelope\r\n   xmlns:SOAP-ENV=\http://schemas.xmlsoap.org/soap/envelope/\\r\n   SOAP-ENV:encodingStyle=\http://schemas.xmlsoap.org/soap/encoding/\>\r\n<SOAP-ENV:Body>\r\n<m:GetAttachDeviceResponse xmlns:m=\urn:NETGEAR-ROUTER:service:DeviceInfo:1\>    <NewAttachDevice></NewAttachDevice>\r\n</m:GetAttachDeviceResponse>\r\n<ResponseCode>501</ResponseCode>\r\n</SOAP-ENV:Body>\r\n</SOAP-ENV:Envelope>\r\n}
Number of attached devices: undefined
First attached device:
null
t = 20.452
trying to get attachedDevices method 2...
Invalid response code from router: 501
{lastResponse:<?xml version=\1.0\ encoding=\UTF-8\?>\r\n<SOAP-ENV:Envelope\r\n   xmlns:SOAP-ENV=\http://schemas.xmlsoap.org/soap/envelope/\\r\n   SOAP-ENV:encodingStyle=\http://schemas.xmlsoap.org/soap/encoding/\>\r\n<SOAP-ENV:Body>\r\n<m:GetAttachDevice2Response xmlns:m=\urn:NETGEAR-ROUTER:service:DeviceInfo:1\>    <NewAttachDevice>\r\n    </NewAttachDevice>\r\n</m:GetAttachDevice2Response>\r\n<ResponseCode>501</ResponseCode>\r\n</SOAP-ENV:Body>\r\n</SOAP-ENV:Envelope>\r\n}
Number of attached devices: undefined
First attached device:
null
t = 24.949
trying to check newFirmware...
check newFirmware method: 2
{currentVersion:V2.3.2.66,newVersion:,releaseNote:}
t = 25.479
trying to logout...
{host:10.0.0.1,port:80,tls:false,username:admin,password:*****,timeout:18000,sessionId:A7D88AE69XXXXXX,cookie:true,loggedIn:false,configStarted:false,soapVersion:3,loginMethod:2,getAttachedDevicesMethod:2,checkNewFirmwareMethod:2,guestWifiMethod:{get50_1:1},lastResponse:...,queue:{queue:{},overheat:1608418352299,pending:true,params:{default:{rule:common,key:common},rules:{common:{rate:3,limit:1,priority:1}},overall:{rate:10,limit:1},retryTime:2,ignoreOverallOverheat:true},heatPart:100}}
t = 25.535
test finished with 5 errors

------------------------------------------------------------------------------------------------------------------------------

2020-12-20 00:59:11 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:58:13 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:57:19 [log] [MyApp] Router compatibility test started
2020-12-20 00:57:12 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:56:11 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:55:11 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:55:04 [netgear] [0] Access control enable requested
2020-12-20 00:55:03 [netgear] [0] Traffic meter enable requested
2020-12-20 00:55:03 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:55:03 [netgear] [0] device init: XR500 id: 5BL1835XXXXXX
2020-12-20 00:54:03 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:53:10 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:52:10 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:51:09 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:50:12 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:49:10 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:49:01 [netgear] [0] Access control enable requested
2020-12-20 00:49:00 [netgear] [0] Traffic meter enable requested
2020-12-20 00:49:00 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:49:00 [netgear] [0] device init: XR500 id: 5BL1835XXXX
2020-12-20 00:48:00 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:47:06 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:46:05 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:45:09 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:44:09 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:43:05 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:3871) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unload listeners added to [Homey]. Use emitter.setMaxListeners() to increase limit
2020-12-20 00:42:58 [netgear] [0] Access control enable requested
2020-12-20 00:42:56 [netgear] [0] Traffic meter enable requested
2020-12-20 00:42:56 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:42:56 [netgear] [0] device init: XR500 id: 5BL1835XXXXXXX
2020-12-20 00:41:56 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:41:02 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:40:01 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:39:02 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:38:01 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:37:02 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:36:55 [netgear] [0] Access control enable requested
2020-12-20 00:36:53 [netgear] [0] Traffic meter enable requested
2020-12-20 00:36:53 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:36:53 [netgear] [0] device init: XR500 id: 5BL1835XXXXX
2020-12-20 00:35:53 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:34:59 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:34:02 [log] [MyApp] test ready
2020-12-20 00:33:59 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:32:58 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:31:59 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:31:11 [log] [MyApp] Router compatibility test started
2020-12-20 00:31:04 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:30:59 [log] [MyApp] Router discovery started
2020-12-20 00:30:52 [netgear] [0] Access control enable requested
2020-12-20 00:30:36 [log] [MyApp] test ready
2020-12-20 00:30:23 [netgear] [0] Traffic meter enable requested
2020-12-20 00:30:23 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:30:23 [netgear] [0] device init: XR500 id: 5BL18XXXXXX
2020-12-20 00:30:13 [netgear] [0] known devices were deleted on request of user
2020-12-20 00:30:13 [netgear] [0] router device settings changed
2020-12-20 00:29:41 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:29:34 [netgear] [0] Access control enable requested
2020-12-20 00:29:32 [netgear] [0] Traffic meter enable requested
2020-12-20 00:29:32 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:29:32 [netgear] [0] device init: XR500 id: 5BL183XXXXXX
2020-12-20 00:28:32 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:27:39 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:27:29 [log] [MyApp] Router compatibility test started
2020-12-20 00:26:38 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:25:38 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:24:37 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:23:38 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:23:31 [netgear] [0] Access control enable requested
2020-12-20 00:23:29 [netgear] [0] Traffic meter enable requested
2020-12-20 00:23:29 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:23:29 [netgear] [0] device init: XR500 id: 5BL183XXXXX
2020-12-20 00:22:29 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:21:35 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:20:35 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:20:28 [log] [MyApp] Retrieving known devices list
2020-12-20 00:19:35 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:18:35 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:17:35 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:17:28 [netgear] [0] Access control enable requested
2020-12-20 00:17:26 [netgear] [0] Traffic meter enable requested
2020-12-20 00:17:26 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:17:26 [netgear] [0] device init: XR500 id: 5BL1835XXXX
2020-12-20 00:16:26 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:15:31 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:14:31 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:14:07 [log] [MyApp] test ready
2020-12-20 00:13:31 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:12:31 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:11:57 [log] [MyApp] Router compatibility test started
2020-12-20 00:11:50 [log] [MyApp] Router discovery started
2020-12-20 00:11:31 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:11:17 [netgear] [0] Access control enable requested
2020-12-20 00:11:16 [netgear] [0] Traffic meter enable requested
2020-12-20 00:11:16 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:11:16 [netgear] [0] device init: XR500 id: 5BL183XXXXX
2020-12-20 00:11:06 [netgear] [0] router device settings changed
2020-12-20 00:10:25 [log] [MyApp] Retrieving known devices list
2020-12-20 00:10:16 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:09:16 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:09:09 [netgear] [0] Access control enable requested
2020-12-20 00:09:08 [netgear] [0] Traffic meter enable requested
2020-12-20 00:09:08 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:09:08 [netgear] [0] device init: XR500 id: 5BL1835XXXXXX
2020-12-20 00:08:14 [log] [MyApp] test ready
2020-12-20 00:08:08 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:07:14 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:06:14 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:05:23 [log] [MyApp] Router compatibility test started
2020-12-20 00:05:13 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:05:03 [log] [MyApp] Router discovery started
2020-12-20 00:04:17 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:03:15 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-20 00:03:07 [netgear] [0] Access control enable requested
2020-12-20 00:03:05 [netgear] [0] Traffic meter enable requested
2020-12-20 00:03:05 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-20 00:03:05 [netgear] [0] device init: XR500 id: 5BL183XXXXXX
2020-12-20 00:02:05 [netgear] [0] watchdog triggered, restarting Homey device now
2020-12-20 00:01:11 [err] [ManagerDrivers] [netgear] [0] DevicePoll Invalid response code from router: 501
2020-12-20 00:00:14 [err] [ManagerDrivers] [netgear] [0] DevicePoll Failed to login
2020-12-19 23:59:27 [err] [ManagerDrivers] [netgear] [0] DevicePoll Failed to login
2020-12-19 23:58:24 [err] [ManagerDrivers] [netgear] [0] DevicePoll socket hang up
2020-12-19 23:57:11 [err] [ManagerDrivers] [netgear] [0] Error: Invalid response code from router: 501 at NetgearRouter._makeRequest (/node_modules/netgear/netgear.js:1691:10) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-19 23:57:03 [netgear] [0] Access control enable requested
2020-12-19 23:57:02 [netgear] [0] Traffic meter enable requested
2020-12-19 23:57:02 [netgear] [0] retrieving knownDevices from persistent storage
2020-12-19 23:57:02 [netgear] [0] router XR500 added as device @ 10.0.0.1:80
2020-12-19 23:57:01 [netgear] [0] device init: XR500 id: 5BL1835XXXXX
2020-12-19 23:56:41 [netgear] Checking router settings from frontend
2020-12-19 23:56:24 [netgear] { Firmware: 'V2.3.2.66WW', RegionTag: 'XR500_WW', Region: 'ww', Model: 'XR500', InternetConnectionStatus: 'Up', ParentalControlSupported: '0', SOAPVersion: '3.18', ReadyShareSupportedLevel: '13', XCloudSupported: '1', LoginMethod: '2.0', host: '10.0.0.1', port: 80 }
2020-12-19 23:56:23 [netgear] discovery started from frontend
2020-12-19 23:56:03 [err] [ManagerDrivers] Error: Cannot find router device in Homey. Router needs to be added first! at EventEmitter. (/drivers/attached_device/driver.js:113:50) at processTicksAndRejections (internal/process/task_queues.js:97:5)
2020-12-19 23:52:32 [netgear] NetgearDriver onInit
2020-12-19 23:52:32 [log_analyzer] LogAnalyzerDriver onInit
2020-12-19 23:52:32 AttachedDeviceDriver onInit
2020-12-19 23:52:32 [log] [MyApp] Netgear App is running!
2020-12-19 23:52:32 [log] [MyApp] capturing stderr
2020-12-19 23:52:32 [log] [MyApp] capturing stdout

**Comment by gruijter** [comment]
Mmh. Interesting. Most commands work ok. But the attached device poll is not working. Do you get the attached devices when you open the netgear nighthawk app?
And did you try to PTP the router already?

**Comment by sebastien-ggp** [comment]
Hello
Thank you for your feedback, yes I did the PTP, I also did a reset without change. I uninstalled everything and then reinstalled without success. I also downgraded two versions with resets still nothing to do the same results. I installed the nightwalk app and indeed the same problem, the peripherals are not present. I reduced the number of peripherals to a minimum to see, nothing to do.Not better with a PTP and reinstall app to homey.Thanks
